packages/admin-sdk/src/iap/index.ts

```diff
@@ -5,8 +5,9 @@ import { createSender } from '../channel';
  */
 export const purchase = createSender('iapCheckout');
 
-export type iapCheckout = {
+export type iapCheckout<T extends 'monthly' | 'yearly' | 'consumable' | 'non-consumable' = 'monthly' | 'yearly' | 'consumable' | 'non-consumable'> = {
     responseType: unknown,
 
     identifier: string,
+    variant: T,
```

Comment:
Since no second property depends on another, you can inline the type. Also, for backwards compatibility, the property shouldn't be required:
```suggestion
export type iapCheckout = {
    responseType: unknown,

    identifier: string,
    variant?: 'monthly' | 'yearly' | 'consumable' | 'non-consumable',
```